### ISSUE TITLE
Doubt: Does ipex.llm.optimize() perform any optimization in run_quantization.py when "qconfig_summary_file" is given

### ISSUE THREAD

**Issue by MadhumithaSrini** [intel/intel-extension-for-pytorch#657]
### Describe the issue

When calling ipex.llm.optimize in line 1036 in https://github.com/intel/intel-extension-for-pytorch/blob/main/examples/cpu/inference/python/llm/single_instance/run_quantization.py, I see that optimize ()'s major blocks are skipped and practically not any optimizations take place.

Is my understanding correct or is there any reason to set the qconfig from static_quant and call ipex.llm.optimize() again? Because already in line 583 (when config_summary_file is provided), we call the ipex.llm.optimize with smooth_quant_config which passes through certain optimization calls in the function already

**Comment by akashdhamasia12** [comment]
Hi, in line 1056: 
if args.benchmark:
    torch._C._jit_set_texpr_fuser_enabled(False)
    qconfig = ipex.quantization.default_static_qconfig_mapping
    user_model = ipex.llm.optimize(
        user_model.eval(),
        dtype=torch.float,
        inplace=True,
        quantization_config=qconfig,
        deployment_mode=False,
    )
    
This block is executed if the --benchmark flag is set. It applies a default static quantization configuration to the model for benchmarking purposes. The deployment_mode is set to False because the model is being prepared for benchmarking, not for actual deployment.

in line 593:
if args.ipex_smooth_quant:
    if args.qconfig_summary_file != "":
        qconfig = ipex.quantization.get_smooth_quant_qconfig_mapping(alpha=args.alpha)
        user_model = ipex.llm.optimize(
            user_model.eval(),
            dtype=amp_dtype,
            quantization_config=qconfig,
            qconfig_summary_file=args.qconfig_summary_file,
            inplace=True,
            deployment_mode=True,
        )
This block of code applies the quantization settings from the qconfig_summary_file to the model and saves the quantized model

In summary, the ipex.llm.optimize() function is used in different contexts within the script:
a. When a qconfig_summary_file is provided, it applies the quantization settings from that file to the model for deployment.
b. When the --benchmark flag is set, it applies a default static quantization configuration to the model for benchmarking purposes.

Please let me know if this resolve your query. Thanks

**Comment by MadhumithaSrini** [comment]
thank you
